tests/unit/ml_handlers/base_ml_test.py

```diff
@@ -0,0 +1,42 @@
+import os
+import time
+import pandas as pd
+
+from mindsdb_sql import parse_sql
+from ..executor_test_base import BaseExecutorTest
+
+
+class BaseMLTest(BaseExecutorTest):
+    """
+    Base test class for ML engines
+    """
+    def wait_predictor(self, project: str, name: str, timeout: int = 100) -> None:
+        """Wait for the predictor to be created, raising an exception if predictor creation fails or exceeds timeout"""
+        for attempt in range(timeout):
+            ret = self.run_sql(f"select * from {project}.models where name='{name}'")
+            if not ret.empty:
+                status = ret["STATUS"][0]
+                if status == "complete":
+                    return
+                elif status == "error":
+                    raise RuntimeError("Predictor failed", ret["ERROR"][0])
+            time.sleep(0.5)
+        raise RuntimeError("Predictor wasn't created")
+
+    def run_sql(self, sql: str) -> pd.DataFrame:
+        """Execute SQL and return a DataFrame, raising an AssertionError if an error occurs"""
+        ret = self.command_executor.execute_command(parse_sql(sql, dialect="mindsdb"))
+        assert ret.error_code is None, f"SQL execution failed with error: {ret.error_code}"
+        if ret.data is not None:
+            columns = [col.alias if col.alias else col.name for col in ret.columns]
+            return pd.DataFrame(ret.data, columns=columns)
+
+
+class BaseMLAPITest(BaseMLTest):
+    """
+    Base test class for API-based ML engines
+    """
+    @staticmethod
+    def get_api_key(env_var: str):
```

Comment:
There is a more robust method for this in "integrations/utilities/handler_utils:get_api_key`, we probably want to use that instead.

Reply:
Understood, @paxcema. I will incorporate that method.

Reply:
Hey @paxcema,
I was just taking a look at the function that you mentioned and I had some thoughts,
1. Since we are running unit tests here, is it correct to assume that we will always be accessing the API keys using environment variables?
2. Are the `create_args` and `engine_storage` parameters relevant here? Because we are running SQL statements to perform our operations?

Based on the above, should this be the way to use the function within `BaseMLAPITest` class?
```
class BaseMLAPITest(BaseMLTest):
    """
    Base test class for API-based ML engines
    """
    @staticmethod
    def get_api_key(api_name: str):
        """Retrieve API key from environment variables"""
        return get_api_key(api_name, {}, None, strict=True)
```
Please let me know what you think, Patricio.

Reply:
Point 1) is actually great. Disregard my original comment and let's keep this to avoid overcomplicating things 😄 sorry for the nuisance